Rich Text Editor | Add label provider in nimble-rich-text-editor (#1459)
ni/nimble: packages/nimble-components/src/label-provider/base/tests/label-providers.mdx

## Core Label Provider

> Provides labels for all Nimble components besides the table and the rich text editor component.

Reviewer: ```suggestion
Provides labels for all Nimble components that do not have a dedicated label provider.
```

Author: Updated

Provides labels for the `nimble-table` and all associated table sub-components / column types.

<Canvas of={tableLabelProvider} sourceState="none" />

> 

Reviewer 2: Let's put the Rich Text Editor before the Table (alphabetical order). That will match the ordering in the tree view, too.

Author: Updated
